sandbox/lua/encoders/schema_influx_write.lua

```diff
@@ -0,0 +1,319 @@
+    of nanoseconds is used to match the precision of Timestamp in Heka messages.
+
+*Example Heka Configuration*
+
+.. code-block:: ini
+
+    [LoadAvgPoller]
+    type = "FilePollingInput"
+    ticker_interval = 5
+    file_path = "/proc/loadavg"
+    decoder = "LinuxStatsDecoder"
+
+    [LoadAvgDecoder]
+    type = "SandboxDecoder"
+    filename = "lua_decoders/linux_loadavg.lua"
+
+    [LinuxStatsDecoder]
+    type = "MultiDecoder"
+    subs = ["LoadAvgDecoder", "AddStaticFields"]
+    cascade_strategy = "all"
+    log_sub_errors = false
+
+    [AddStaticFields]
+    type = "ScribbleDecoder"
+
+        [AddStaticFields.message_fields]
+        Environment = "DEV"
+
+    [InfluxdbOutput]
+    type = "HttpOutput"
+    message_matcher = "Type =~ /stats.*/"
+    encoder = "InfluxdbEncoder"
+    address = "http://influxdbserver.example.com:8086/write"
+    username = "influx_username"
+    password = "influx_password"
+
+    [InfluxdbEncoder]
+    type = "SandboxEncoder"
+    filename = "lua_encoders/schema_influx_write.lua"
+
+        [InfluxdbEncoder.config]
+        name_prefix = "%{Hostname}.%{Type}"
+        skip_fields = "**all_base** FilePath NumProcesses Environment"
+        tag_fields = "Hostname Environment"
+        timestamp_precision= "s"
+
+*Example Output*
+
+.. code-block:: json
+
+    {"database":"mydb","retentionPolicy":"","tags":{"Environment":"DEV","Hostname":"my_hostname"}, "points":[{"name":"my_hostname.stats.loadavg.5MinAvg","fields":{"value":0.03}},{"name":"my_hostname.stats.loadavg.15MinAvg","fields":{"value":0.05}},{"name":"my_hostname.stats.loadavg.1MinAvg","fields":{"value":0.01}}],"timestamp":1426439735,"precision":"s"}
+
+--]=]
+
+require "cjson"
+require "math"
+require "string"
+
+local name_prefix_orig  = read_config("name_prefix") or "name"
+local name_prefix = name_prefix_orig
+local use_subs
+if string.find(name_prefix, "%%{[%w%p]-}") then
+    use_subs = true
+end
+
+-- Default this option to "mydb"
+local influxdb_db = read_config("database") or "mydb"
+
+-- Default this option to ""
+local retention_policy = read_config("retention_policy") or ""
+
+-- Default this option to "ms"
+local timestamp_precision = read_config("timestamp_precision") or "ms"
+
+local base_fields_map = {
+    Type = true,
+    Payload = true,
+    Hostname = true,
+    Pid = true,
+    Logger = true,
+    Severity = true,
+    EnvVersion = true
+}
+
+local base_fields_list = {
+    "Type",
+    "Payload",
+    "Hostname",
+    "Pid",
+    "Logger",
+    "Severity",
+    "EnvVersion"
+}
+
+local base_fields_tag_map = {
+    Type = true,
+    Hostname = true,
+    Severity = true,
+    Logger = true
+}
+
+-- Used for interpolating message fields into name_prefix.
+local function sub_func(key)
+    if base_fields_map[key] then
+        return read_message(key)
+    else
+        local val = read_message("Fields["..key.."]")
+        if val then
+            return val
+        end
+        return "%{"..key.."}"
+    end
+end
```

Comment:
It was actually @RobCherry who did the work of teasing that stuff out... I assumed he was working with you, but I guess not? Anyway, thanks to both of you for helping out! :)